src/modules/dashboard/components/ColonyHomeActions/ColonyHomeActions.tsx

```diff
@@ -79,6 +80,13 @@ const ColonyHomeActions = ({ colony, ethDomainId }: Props) => {
     variables: { address: colony.colonyAddress },
   });
 
+  const {
+    data: colonyMembers,
+    loading: isLoadingColonyMembers,
+  } = useMembersSubscription({
+    variables: { colonyAddress: colony.colonyAddress },
+  });
```

Comment:
First of all, the approach should be similar in all cases - both for Smite/Award dialogs and for Create payment dialog which is not the case right now. There is no reason for the approaches to be different.

Secondly, I think that the query shouldn't be moved here. Will we now move all the queries from all of the dialogs here? It doesn't make sense. I think that it should removed from here and the same approach should be takes as for the Create payment dialog.

Reply:
@chinins The intention here is to ensure that `colonyMembers` is defined when the component mounts. If it isn't, as is the case when you run the query in the component, you can get a slight flashing in the UI, as below (see recipient field): 

![flashingRep](https://user-images.githubusercontent.com/64402732/185097273-5bf9c640-f295-488b-b429-ca0b0e8bd0c2.gif)

If the colonyMembers data is undefined, then the selectedUser will also be undefined. Once the query returns, the component rerenders and the selectedUser gets set properly -- hence the flash. 

The reason this wouldn't be necessary in the CreatePayment dialog is because no user is preselected, so the effect of colonyMembers being undefined initially is invisible -- by the time a user has opened the SingleUserPicker, the data is there. 

What do you think? I'm easy either way, but in general prefer to avoid UI flashes, even small ones.

Reply:
To be honest, I don't mind this flash. I think we shouldn't be add queries in improper places just to avoid it. You can have a loading spinner instead for the whole dialog if you want. And then when we move to the cdapp with new react features there either won't be any flashing at all, just a tiny delay in showing the form, or you will have a spinner for a couple of seconds.